platform/security/sts/security-sts-server/src/test/java/ddf/security/sts/SubjectDNConstraintsInterceptorTest.java

```diff
@@ -0,0 +1,126 @@
+    }
+
+    @Test
+    public void testHandleMessageCommaPattern() {
+        Message message = mock(Message.class);
+        when(message.get(WSHandlerConstants.SIG_SUBJECT_CERT_CONSTRAINTS)).thenReturn(
+                "CN=cert,OU=unit|CN=stuff,OU=more");
+        Request request = mock(Request.class);
+        when(message.get("HTTP.REQUEST")).thenReturn(request);
+        X509Certificate x509Certificate = mock(X509Certificate.class);
+        X500Principal x500Principal = new X500Principal("CN=cert,OU=unit");
+        when(x509Certificate.getSubjectX500Principal()).thenReturn(x500Principal);
+        when(request.getAttribute("javax.servlet.request.X509Certificate")).thenReturn(new X509Certificate[] {
+                x509Certificate});
+
+        subjectDNConstraintsInterceptor.handleMessage(message);
+    }
+
+    @Test
+    public void testHandleMessageNoPattern() {
+        Message message = mock(Message.class);
+        when(message.get(WSHandlerConstants.SIG_SUBJECT_CERT_CONSTRAINTS)).thenReturn(null);
+        Request request = mock(Request.class);
+        when(message.get("HTTP.REQUEST")).thenReturn(request);
+        X509Certificate x509Certificate = mock(X509Certificate.class);
+        X500Principal x500Principal = new X500Principal("CN=cert");
+        when(x509Certificate.getSubjectX500Principal()).thenReturn(x500Principal);
+        when(request.getAttribute("javax.servlet.request.X509Certificate")).thenReturn(new X509Certificate[] {
+                x509Certificate});
+
+        subjectDNConstraintsInterceptor.handleMessage(message);
+    }
+
+    @Test(expected = AccessDeniedException.class)
+    public void testHandleMessageNoMatch() {
+        Message message = mock(Message.class);
+        when(message.get(WSHandlerConstants.SIG_SUBJECT_CERT_CONSTRAINTS)).thenReturn("CN=cert");
+        Request request = mock(Request.class);
+        when(message.get("HTTP.REQUEST")).thenReturn(request);
+        X509Certificate x509Certificate = mock(X509Certificate.class);
+        X500Principal x500Principal = new X500Principal("CN=certbad");
+        when(x509Certificate.getSubjectX500Principal()).thenReturn(x500Principal);
+        when(request.getAttribute("javax.servlet.request.X509Certificate")).thenReturn(new X509Certificate[] {
+                x509Certificate});
+
+        subjectDNConstraintsInterceptor.handleMessage(message);
+    }
+
+    @Test(expected = AccessDeniedException.class)
+    public void testHandleMessageNoCert() {
+        Message message = mock(Message.class);
+        when(message.get(WSHandlerConstants.SIG_SUBJECT_CERT_CONSTRAINTS)).thenReturn("CN=certbad");
+        Request request = mock(Request.class);
+        when(message.get("HTTP.REQUEST")).thenReturn(request);
+        when(request.getAttribute("javax.servlet.request.X509Certificate")).thenReturn(null);
+
+        subjectDNConstraintsInterceptor.handleMessage(message);
+    }
+
+    @Test
+    public void testSetSubjectDNPatterns() {
```

Comment:
✏️  There's a lot of repeated setup code in this test class. We should consider extracting that code to a private method that takes parameters, use JUnit parametrized tests, or use Spock. Going with parametrized tests or Spock would help capture all the pattern tests in `testSetSubjectDNPatterns` (which really should be separate tests) and not have to expose the `setSubjectDNPatterns` method.